trieste/objectives/multi_objectives.py

```diff
@@ -236,3 +248,66 @@ def gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
         search_space=Box([0.0], [1.0]) ** d,
         gen_pareto_optimal_points=gen_pareto_optimal_points,
     )
+
+
+def ConstrainedBraninCurrin() -> ConstrainedMultiObjectiveTestProblem:
+    """
+    The ConstrainedBraninCurrin problem, typically evaluated over :math:`[0, 1]^2`.
+    See :cite:`belakaria2019max` and :cite:`daulton2020differentiable`
+    (the latter for adding the constraint) for details.
+
+    :return: The problem specification.
+    """
+
+    def gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
+        return tf.zeros(shape=0, dtype=tf.float64)
```

Comment:
so there is no known pareto front for this problem I take it?
can you please add a note to the docs of the problem above stating that and that this function will return an empty tensor please

Reply:
@uri-granta do you think this is ok or we should return `NotImplementedError`? for some problems pareto front will not be available

it's a bit burdensome, but what can be done is to compute offline approximate ground truth with say genetic algos with a large number of function evaluations, store a fixed number of points in the repo as a text file and then load that when the function is called

Reply:
@TsingQAQ can you please raise an error here, but perhaps in the docs you can indicate to user a way to generate approximate ground truth

Reply:
Sure! for readability have created a new exception class [`NoAnalyticalParetoPointsError`](https://github.com/TsingQAQ/trieste/blob/e0ac5d45f649aef832badd6c7855fb829b92fd7c/trieste/objectives/multi_objectives.py#L32), please check if this is idea.